### ISSUE TITLE
Uploading a file crashes with NME (MiniMime and v2.15.0)

### ISSUE THREAD

**Issue by jdashton** [teamcapybara/capybara#1898]
## Meta
Capybara Version: 2.15.0
Driver Information (and browser if relevant):

## Expected Behavior
Tests run to completion

## Actual Behavior
```
NoMethodError: undefined method `content_type' for nil:NilClass
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/rack_test/form.rb:45:in `block in params'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/nokogiri-1.8.0/lib/nokogiri/xml/node_set.rb:190:in `block in each'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/nokogiri-1.8.0/lib/nokogiri/xml/node_set.rb:189:in `upto'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/nokogiri-1.8.0/lib/nokogiri/xml/node_set.rb:189:in `each'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/rack_test/form.rb:28:in `map'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/rack_test/form.rb:28:in `params'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/rack_test/form.rb:77:in `submit'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/rack_test/node.rb:64:in `click'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/node/element.rb:143:in `block in click'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/node/base.rb:85:in `synchronize'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/node/element.rb:143:in `click'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/node/actions.rb:61:in `click_button'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/session.rb:776:in `block (2 levels) in <class:Session>'
/Users/jdashton/.rvm/gems/ruby-2.3.4@arp_upload_20170727/gems/capybara-2.15.0/lib/capybara/dsl.rb:50:in `block (2 levels) in <module:DSL>'
./spec/features/event_creation_spec.rb:39:in `block (2 levels) in <top (required)>'
-e:1:in `load'
-e:1:in `<main>'
```
## Steps to reproduce

This happens at line 45 in `form.rb`:
```ruby
  Rack::Test::UploadedFile.new(value, MiniMime.lookup_by_filename(value).content_type)
```

We're trying to submit (emulate a file upload of) a custom file when we hit this line, and this file has no standard MIME type.

I poked around looking for release notes, but didn't look in the right place to find them. If something needs to be adjusted to make old tests work under MiniMime, the release notes would be a nice place to mention the necessary changes.

But I give you Kudos on pushing a new release out this door this morning. Congrats!

**Comment by duffyjp** [comment]
I'm getting the same thing with any file with no file extension.  I have a bunch of specs where I upload my `Gemfile` just because it's always going to be there.

```ruby
attach_file("uploads[]", 'Gemfile')
```
These all fail on 2.15.0

**Comment by twalpole** [comment]
Whoops -- sorry about that - 2.15.1 will be coming as soon as I've fixed it

**Comment by twalpole** [comment]
Just waiting for all the tests to pass

**Comment by twalpole** [comment]
Fixed with the release of 2.15.1

**Comment by jdashton** [comment]
Confirmed. Thank you for the swift fix!
